src/client/api/events/sub_outputs_payload.c

```diff
@@ -4,108 +4,122 @@
 #include "client/api/events/sub_outputs_payload.h"
 #include "client/api/json_parser/json_utils.h"
 #include "core/address.h"
+#include "core/models/outputs/output_foundry.h"
+#include "core/utils/iota_str.h"
+#include "core/utils/macros.h"
 
-int event_parse_outputs_payload(char const data[], event_outputs_payload_t *res) {
-  int ret = -1;
-  cJSON *json_obj = cJSON_Parse(data);
-  if (json_obj == NULL) {
-    printf("[%s:%d] OOM\n", __func__, __LINE__);
-    return ret;
+int event_sub_outputs_id(event_client_handle_t client, int *mid, char const output_id[], int qos) {
+  if ((strlen(output_id)) != BIN_TO_HEX_BYTES(IOTA_OUTPUT_ID_BYTES)) {
```

Comment:
One parenthesis too many.
```suggestion
  if (strlen(output_id) != BIN_TO_HEX_BYTES(IOTA_OUTPUT_ID_BYTES)) {
```